src/weatherdata/ipm.py

```diff
@@ -78,9 +78,26 @@ def list_available_parameters(self):
            parameters = values[self.name]
 
         return parameters
+
+    def endpoint(self):
+        """
+        Get endpoint associate at the name parameter of WeatherDataSource
+
+        Parameters:
+        -----------
+
+        Returns:
+        --------
+            a endpoint (str) used in get_data function
+        """
+        endpoints = self.ipm.weatheradapter_service()
+
+        if self.name in endpoints:
+            endpoint = endpoints[self.name]
         
+        return endpoint    
          
-    def get_data(self,parameters, station_id, daterange):
+    def get_data(self,parameters=[1002,3002], station_id=101104, daterange=pandas.date_range('2020-06-12T00:00:00','2020-07-03T00:00:00',freq='H',tz="UTC")):
```

Comment:
- Put the def of the function on multiple line.
- Simplify the date range.
- Where these parameters come from? Can they be obtain by query?